src/nvim/regexp_defs.h

```diff
@@ -32,7 +32,16 @@
 
 // In the NFA engine: how many states are allowed.
 #define NFA_MAX_STATES 100000
-#define NFA_TOO_EXPENSIVE -1
+
+/// Result of a regex search.
+///
+/// kRegexMatchNotFound and kRegexMatchesFound may also be used for their
+/// integer values.
```

Comment:
We generally discourage this--isn't it equivalent to just using `true`/`false` instead of an enum? Did you make this choice to avoid cascading changes?